spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/StreamListenerTestHandlerBean.java

```diff
@@ -0,0 +1,140 @@
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Output;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.handler.annotation.SendTo;
+import org.springframework.util.MimeType;
+import org.springframework.util.MimeTypeUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(Parameterized.class)
+public class StreamListenerTestHandlerBean {
+
+	private Class<?> configClass;
+
+	public StreamListenerTestHandlerBean(Class<?> configClass) {
+		this.configClass = configClass;
+	}
+
+	@Parameterized.Parameters
+	public static Collection InputConfigs() {
+		return Arrays.asList(new Class[] { TestHandlerBean1.class, TestHandlerBean2.class });
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testHandlerBean() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(this.configClass,
+				"--spring.cloud.stream.bindings.output.contentType=application/json",
+				"--server.port=0");
+		MessageCollector collector = context.getBean(MessageCollector.class);
+		Processor processor = context.getBean(Processor.class);
+		String id = UUID.randomUUID().toString();
+		processor.input().send(
+				MessageBuilder.withPayload("{\"foo\":\"barbar" + id + "\"}")
+						.setHeader("contentType", "application/json").build());
+		HandlerBean handlerBean = context.getBean(HandlerBean.class);
+		assertThat(handlerBean.receivedPojos).hasSize(1);
+		assertThat(handlerBean.receivedPojos.get(0)).hasFieldOrPropertyWithValue("foo",
+				"barbar" + id);
+		Message<String> message = (Message<String>) collector.forChannel(
+				processor.output()).poll(1, TimeUnit.SECONDS);
+		assertThat(message).isNotNull();
+		assertThat(message.getPayload()).isEqualTo("{\"bar\":\"barbar" + id + "\"}");
+		assertThat(message.getHeaders().get(MessageHeaders.CONTENT_TYPE, MimeType.class)
+				.includes(MimeTypeUtils.APPLICATION_JSON));
+		context.close();
+	}
+
+	@EnableBinding(Processor.class)
+	@EnableAutoConfiguration
+	public static class TestHandlerBean1 {
+
+		@Bean
+		public HandlerBean1 handlerBean() {
+			return new HandlerBean1();
+		}
+	}
+
+	@EnableBinding(Processor.class)
+	@EnableAutoConfiguration
+	public static class TestHandlerBean2 {
+
+		@Bean
+		public HandlerBean2 handlerBean() {
+			return new HandlerBean2();
+		}
+	}
+
+	public static class HandlerBean1 extends HandlerBean {
```

Comment:
They are identical except one uses `@Output` while the other uses `@SendTo` when specifying the `outbound` channel name :-)